modules/playlists/src/main/java/org/opencastproject/playlists/PlaylistRestService.java

```diff
@@ -0,0 +1,600 @@
+      + "<description>This is a playlist about Opencast</description><creator>Opencast</creator>"
+      + "<updated>1701787700848</updated><accessControlEntries><allow>true</allow><role>ROLE_USER_BOB</role>"
+      + "<action>read</action></accessControlEntries></ns3:playlist>";
+
+  public static final String SAMPLE_PLAYLIST_ENTRIES_XML =
+      "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\"?>\n"
+      + "<entries>\n"
+      + "\t<entry id=\"1061\">\n" + "\t\t<contentId>ID-av-portal</contentId>\n" + "\t\t<type>EVENT</type>\n"
+      + "\t</entry>\n"
+      + "\t<entry id=\"1062\">\n" + "\t\t<contentId>ID-av-print</contentId>\n" + "\t\t<type>EVENT</type>\n"
+      + "\t</entry>\n" + "</entries>";
+
+  /** The playlist service instance */
+  private PlaylistService service;
+
+  /** The search service */
+  protected SearchService searchService = null;
+
+  /** The authorization service */
+  protected AuthorizationService authorizationService = null;
+
+  /**
+   * Sets the playlist service
+   *
+   * @param service
+   *          the playlist service instance
+   */
+  @Reference
+  public void setService(PlaylistService service) {
+    this.service = service;
+  }
+
+  @Reference
+  protected void setSearchService(SearchService searchService) {
+    this.searchService = searchService;
+  }
+
+  @Reference
+  public void setAuthorizationService(AuthorizationService authorizationService) {
+    this.authorizationService = authorizationService;
+  }
+
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  @Path("playlist.json")
+  @RestQuery(
+      name = "playlist",
+      description = "Get a playlist.",
+      returnDescription = "A playlist as JSON",
+      restParameters = {
+          @RestParameter(name = "id", isRequired = true, description = "The playlist identifier", type = STRING),
+          @RestParameter(name = "withPublications", isRequired = false, description = "If available publications for"
+              + "the content should be returned. Only works for content of type EVENT.", type = BOOLEAN,
+              defaultValue = "true")
+      },
+      responses = {
+          @RestResponse(responseCode = SC_OK, description = "A playlist as JSON."),
+          @RestResponse(responseCode = SC_NOT_FOUND, description = "No playlist with that identifier exists."),
+          @RestResponse(responseCode = SC_UNAUTHORIZED, description = "Not authorized to perform this action")
+      })
+  public JaxbPlaylist getPlaylistAsJson(
+      @FormParam("id") String id,
+      @FormParam("withPublications") boolean withPublications)
+          throws NotFoundException, UnauthorizedException {
+    Playlist playlist = service.getPlaylistById(id);
+
+    JaxbPlaylist jaxbPlaylist;
+    if (withPublications) {
+      jaxbPlaylist = service.enrich(playlist);
+    } else {
+      jaxbPlaylist = new JaxbPlaylist(playlist);
+    }
+
+    return jaxbPlaylist;
+  }
+
+  @GET
+  @Produces(MediaType.APPLICATION_XML)
+  @Path("playlist.xml")
```

Comment:
Since I did not know which representation people use/would want to use, I employed Jaxb as a means to get XML "for free" along with JSON. Turns out that wasn't as free as I originally hoped, but oh well, hindsight is 20/20. I guess we could remove the whole Jaxb stuff in the future if really no one cares for XML.